src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -273,10 +273,10 @@ - (void)invalidateCASESession
 class SubscriptionCallback final : public MTRBaseSubscriptionCallback {
 public:
     SubscriptionCallback(dispatch_queue_t queue, DataReportCallback attributeReportCallback, DataReportCallback eventReportCallback,
-        ErrorCallback errorCallback, SubscriptionEstablishedHandler _Nullable subscriptionEstablishedHandler,
-        OnDoneHandler _Nullable onDoneHandler)
-        : MTRBaseSubscriptionCallback(
-            queue, attributeReportCallback, eventReportCallback, errorCallback, nil, subscriptionEstablishedHandler, onDoneHandler)
+        ErrorCallback errorCallback, MTRDeviceResubscriptionScheduledHandler _Nullable resubscriptionScheduledHandler,
+        SubscriptionEstablishedHandler _Nullable subscriptionEstablishedHandler, OnDoneHandler _Nullable onDoneHandler)
+        : MTRBaseSubscriptionCallback(queue, attributeReportCallback, eventReportCallback, errorCallback,
+            resubscriptionScheduledHandler, subscriptionEstablishedHandler, onDoneHandler)
```

Comment:
It is a bit counter intuitive that params order for `resubscriptionScheduledHandler` and `subscriptionEstablishedHandler` are reversed between `SubscriptionCallback` and `MTRBaseSubscriptionCallback`

Reply:
I'm not following.  The callbacks both have resubscription, then established.

The order is different for the public subscribeWithQueue API, where resubscription got added last.  I _could_ add it before subscriptionEstablished there, but was trying to minimize API churn.

Reply:
I guess I meant that the SubscriptionCallback constructor in `src/darwin/Framework/CHIP/MTRDevice.mm` is:
```
class SubscriptionCallback final : public MTRBaseSubscriptionCallback {
 public:
     SubscriptionCallback(dispatch_queue_t queue, DataReportCallback attributeReportCallback, DataReportCallback eventReportCallback,
         ErrorCallback errorCallback, SubscriptionEstablishedHandler subscriptionEstablishedHandler,
         MTRDeviceResubscriptionScheduledHandler resubscriptionCallback, OnDoneHandler onDoneHandler)
         : MTRBaseSubscriptionCallback(queue, attributeReportCallback, eventReportCallback, errorCallback, resubscriptionCallback,
             subscriptionEstablishedHandler, onDoneHandler)
     {
     }
```
where things are reversed. Not in  `src/darwin/Framework/CHIP/MTRBaseDevice.mm`
It was before your changes but it looks weird :confused:

Reply:
Adjusted that ordering bit.